### PR TITLE
Use generator

### PR DESCRIPTION
**Pull request by ambar**
[gogen](/ambar/gogen) is a 36K generator, which is suitable for scaffold projects.

Once this project is released, we can create project like this:

```sh
npx postcss-plugin-boilerplate <directory>
```

Or, you can try it before the release:

```sh
# not working now ~~npx gogen ambar/postcss-plugin-boilerplate myplugin~~

# test current branch
npx postcss-plugin-boilerplate@ambar/postcss-plugin-boilerplate#use-generator test
```

### CODE REVIEW

**Comment by ambar** [comment]
It's easier to test and maintain (use a separate template directory), less time from start to development (no clone/start script)

File: template/README.t.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
 # PLUGIN_TITLE
```
Comment (ai) [inline template/README.t.md:1]: Forgotten template

File: template/README.t.md [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
 
-[PostCSS] plugin PLUGIN_DESC.
-
-[PostCSS]: https://github.com/postcss/postcss
+${description}.
```
Comment (ai) [inline template/README.t.md:3]: Why we changed description format and removed link to PostCSS?

File: template/gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+node_modules/
+.DS_Store
```
Comment (ai) [inline template/gitignore:2]: System files should be in user’s git ignore in my system

File: template/README.t.md (hunk 2 above)
Comment (ambar) [reply template/README.t.md:3]: It's the restriction from `prompts`, I can't figure out a way to preserve the leading words.

https://github.com/terkelg/prompts#options

All others are fixed.

**Comment by lumburr** [comment]
> [gogen](/ambar/gogen) is a 36K generator, which is suitable for scaffold projects.
> 
> Once this project is released, we can create project like this:
> 
> ```shell
> npx postcss-plugin-boilerplate <directory>
> ```
> 
> Or, you can try it before the release:
> 
> ```shell
> npx gogen ambar/postcss-plugin-boilerplate myplugin
> ```
I run`npx gogen ambar/postcss-plugin-boilerplate myplugin` on window.

屏幕截图 2021-10-13 103338
I took a look at `gogen`, maybe using `exec` instead of `spawn` is more compatible with different systems.

**Comment by ambar** [comment]
@lumburr Thanks for feedback, the `touch` was added for yarn berry, replaced with fs.writeFile now: https://github.com/ambar/gogen/commit/2d1e9be38c081d3c08af1757d990263f67d57dd3

File: template/README.t.md (hunk 2 above)
Comment (ambar) [reply template/README.t.md:3]: Replace with `enquirer` snippet, which is perfect for this: https://github.com/enquirer/enquirer#snippet-prompt

**Comment by ambar** [comment]
It's `@ambar_li`, thanks for merging.

**Comment by lumburr** [comment]
Fantastic!

**Pull request merged**